flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java

```diff
@@ -346,9 +346,6 @@ private MemorySegment releaseSegmentsForOwnerUntilNextOwner(
                     while (segmentsIterator.hasNext()) {
                         MemorySegment segment = segmentsIterator.next();
                         try {
-                            if (segment == null || segment.isFreed()) {
```

Comment:
I think removing of `segment == null` could be in a dedicated cleanup jira. I guess `null` `MemorySegment` works in existing code on purpose though I don't think it is a good.

Reply:
Not sure about this. I don't see from the codes how `segment` can possibly be `null` atm., and the CI also proves it. I'd rather not opening a dedicated ticket for it.

Reply:
I am also not sure. But old code passes following testing code, while new code fails.

```java
    @Test
    public void testMemoryManagerReleaseNullInMedian() throws Exception {
        List<MemorySegment> segments = memoryManager.allocatePages(this, 5);
        segments.add(null);
        memoryManager.allocatePages(this, segments, 5);
        memoryManager.release(segments);
    }
```

I raised this solely because explicit null-tolerance in `MemoryManager.release` not my preference. If we want drop null-tolerance, I think a dedicated follow up is needed anyway. Otherwise, following code still passes in new code.

```java
    @Test
    public void testMemoryManagerReleaseNullInHead() throws Exception {
        List<MemorySegment> segments = new ArrayList<>();
        segments.add(null);
        memoryManager.allocatePages(this, segments, 5);
        memoryManager.release(segments);
    }
```

A dedicated follow up express strong opinion to drop null-tolerance.

I am ok with current situation if there is a consensus and a followup to drop this null-tolerance. Otherwise, I think it introduce inconsistent in codes.

Reply:
@kezhuw,
Thanks for the explanation. Now I see how could the `segment` be `null`.

I think it makes sense for `release(Collection<MemorySegment>)` being able to handle potential `null` values in the argument collection, either ignore it or fail. But I have to say its super implicit that the iterator passed into `releaseSegmentsForOwnerUntilNextOwner()` still contains null values.

Anyway, I agree that it indeed deserves a dedicated ticket to gracefully handle the null values. I'll revert the removal of null check.

Reply:
I think the follow up should be either dropping null-tolerance or adding test case to guard null-tolerance. The implementation is carefully designed(I guess so) while there is no test to assert on.